### PR TITLE
fix(DeliveryRequestApiService): filter deliveries requested by partner role and incoterms

### LINKED ISSUE #432: Customer can't refresh Delivery Information
## Current behaviour

Currently [`DeliveryRequestApiService.handleDeliverySubmodelRequest`](https://github.com/eclipse-tractusx/puris/blob/main/backend/src/main/java/org/eclipse/tractusx/puris/backend/delivery/logic/service/DeliveryRequestApiService.java#L88) always aborts mapping of internal delivery information to a partner, if the partner is not a supplier. 

Likely the check has been introduced to steer the direction of the two-sided information without a dedicated 

Note: We added the check during #382. Further checking than previously makes sense, but was not sufficient / partly wrong.

## Expected behavior

Depending on the Incoterm and the partner relationship the delivery information needs to be validated.

The [`IncotermEnumeration`](https://github.com/eclipse-tractusx/puris/blob/9595354197b6a2d9976b45ca390fb0e7fc5c46e5/backend/src/main/java/org/eclipse/tractusx/puris/backend/delivery/domain/model/IncotermEnumeration.java#L54) already lists the responsibility:
- `CUSTOMER` = Customer creates the delivery information. Supplier asks for it
- `SUPPLIER` = Supplier creates the delivery information. Customer asks for it
- `PARTIAL` = both applies

## Steps to reproduce the Bug

- Start applications (no role)
- use the postman collection for integration test to seed data
- try to refresh data on the dashboard as a customer from the supplier

### PR DESCRIPTION
**Pull request by tom-rm-meyer-ISST**
## Description

resolves #432 

Maybe we need also to do similar enhancements to the following fields
- `DeliveryService` / `OwnDeliveryService`
- Frontend > incoterm and partner relationship including current role in dashboard (not sure, if we already have it but I think there is something in place)

Please let me know what you think. Based on the changes I would release a patch version

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

**Review by ReneSchroederLJ — CHANGES_REQUESTED** [review]
Thank you for your contribution. 

Overall solid work, but besides a comment I left on a bug there seems to be an issue with how the material is retrieved in `handleDeliverySubmodelRequest`. There is a disconnect between the comment and what is actually happening. 

At first we attempt to find the material by assuming that we are supplier and the Cx number is our own. If this fails we should check if we are customer and the Cx number matches the partnerCxNumber in any mpr for that partner. only if that fails we should `triggerPartTypeRetrieval` and afterwards check the mpr again.

Currently we do not check the mpr at all so we can not handle requests where we are the customer.

**Comment by tom-rm-meyer-ISST** [comment]
> Thank you for your contribution.
> 
> Overall solid work, but besides a comment I left on a bug there seems to be an issue with how the material is retrieved in `handleDeliverySubmodelRequest`. There is a disconnect between the comment and what is actually happening.
> 
> At first we attempt to find the material by assuming that we are supplier and the Cx number is our own. If this fails we should check if we are customer and the Cx number matches the partnerCxNumber in any mpr for that partner. only if that fails we should `triggerPartTypeRetrieval` and afterwards check the mpr again.
> 
> Currently we do not check the mpr at all so we can not handle requests where we are the customer.

Well seen! While implementing the change and a test to do it I recognized that this might even be unwanted behavior: If we don't have the CX ID when being customer, this means we did not create a Shell Descriptor with a href. Someone tries to access data directly via the asset / interface without the DTR to see how to access the data. 

I added a warning message as I think this could be a security issue, if a partner tries to do that.

File: backend/src/main/java/org/eclipse/tractusx/puris/backend/delivery/logic/service/DeliveryRequestApiService.java [hunk 1]
```diff
@@ -126,4 +160,38 @@ public void doReportedDeliveryRequest(Partner partner, Material material) {
...
+                // unlikely that we have a mpr without any role set but then we also don't return something
+                if (mpr.isPartnerBuysMaterial() && mpr.isPartnerSuppliesMaterial()) {
+                    return false;
+                } else return mpr.isPartnerBuysMaterial() || mpr.isPartnerSuppliesMaterial();
+            }
+
+            // if we have incoterms set, filter more sophisticatedly based on responsbility of incoterms and role
+            // derived from material number cx
+            DeliveryResponsibilityEnumeration responsibility = delivery.getIncoterm().getResponsibility();
+            if (partnerIsCustomer) {
```
[the remark this answers is not in the document]
Comment (tom-rm-meyer-ISST) [reply backend/src/main/java/org/eclipse/tractusx/puris/backend/delivery/logic/service/DeliveryRequestApiService.java:188]: Great catch! updated + tests updated

**Review by ReneSchroederLJ — APPROVED** [review]
Looks good to me. Thank you for your efforts.

**Comment by tom-rm-meyer-ISST** [comment]
Bumped version to create a patch release

**Review by mhellmeier — APPROVED** [review]

**Pull request merged**